src/components/Searcher.js

```diff
@@ -970,6 +989,44 @@ class Searcher extends React.Component<SearcherDefaultProps, SearcherProps, Sear
     });
   }
 
+  /**
+   * Add signal each time a filter is added/removed.
+   * When a filter is added, signal with weight 1 is created.
+   * When a filter is removed, signal with weight 0 is created.
+   */
+  addFacetFilterSignal(facetFilter: FacetFilter, weight: number) {
+    const queryDocuments = this.state.response ? this.state.response.documents : null;
+    const querySignal = queryDocuments ? queryDocuments[0].signal : null;
+    const facets = this.state.response ? this.state.response.facets : null;
+    if (!querySignal || !facets) {
+      return;
+    }
+    const facet = facets.find((searchFacet) => {
+      return searchFacet.label === facetFilter.facetName;
+    });
+    if (!facet) {
+      return;
+    }
+    const signal = new SignalData();
+    signal.docId = facetFilter.filter;
+    signal.docOrdinal = facet.buckets.findIndex((bucket) => {
+      return bucket.filter === facetFilter.filter;
+    }) + 1; // index starts at 1
+    signal.featureVector = '';
+    signal.locale = querySignal.locale;
+    signal.principal = querySignal.principal;
+    signal.query = querySignal.query;
+    signal.queryTimestamp = querySignal.queryTimestamp;
+    signal.relevancyModelName = querySignal.relevancyModelName;
+    signal.relevancyModelNames = querySignal.relevancyModelNames;
+    signal.relevancyModelVersion = querySignal.relevancyModelVersion;
+    signal.signalTimestamp = Date.now();
+    signal.ttl = false;
+
+    const doc = new SearchDocument(new Map(), signal);
+    new Signals(this.props.baseUri).addSignal(doc, 'facet', weight);
```

Comment:
Actually, if we're going to make a breaking change, we should probably make the method static and pass the baseUri to it as a parameterl so you don't have to do the `new Signals()` part... just call `Signals.addSignal(baseUri, signal, type, weight)`.
How does this sound?

Reply:
Sounds good. We could in fact create a new function that accepts the Signal object directly. This would ensure the original `addSignal()` still works. For signals of types `like` and `click', the expects to send a `SearchDocument` from which the `Signal` object would be extracted. I think we should still support the existing `addSignal()` which is used for these signal types, and maybe any other in the future as well?

Reply:
I just took a quick peek in the Attivio GitHub and there are actually a couple of cases where people outside of SUIT are calling this method... so let's leave it intact for now...

And just add a second method... maybe called `addRawSignal` that takes just the signal object and no type or weight. The places where you're currently creating an empty SearchDocument in order to add a signal should call this new method instead...